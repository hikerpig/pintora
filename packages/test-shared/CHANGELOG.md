# Change Log

## 0.2.0-alpha.3

### Patch Changes

- bc16cae: [gantt] add support for axisInterval
- 46a1a9f: [gantt] add support for axisInterval

## 0.2.0-alpha.2

### Patch Changes

- 235cc0e: add @config and @param to gantt diagram; update tmLanguage;

## 0.2.0-alpha.1

### Minor Changes

- bd214c7: Add gantt diagram

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2-beta.0](https://github.com/hikerpig/pintora/compare/v0.1.1...v0.1.2-beta.0) (2022-02-13)

### Bug Fixes

- should support `\r` as end-of-line char and whitespace char ([f0696f4](https://github.com/hikerpig/pintora/commit/f0696f48dcb2e82b7368b8a2908dc938b6c3aea2))

# [0.1.0](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.27...v0.1.0) (2022-02-06)

**Note:** Version bump only for package @pintora/test-shared

# [0.1.0-alpha.24](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.23...v0.1.0-alpha.24) (2022-02-01)

### Features

- **pintora-diagrams:** [componentDiagram] we can have relationship from element to group ([9cc74cd](https://github.com/hikerpig/pintora/commit/9cc74cdc53013b6b63a13658ff719d414d04706f))

# [0.1.0-alpha.19](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.18...v0.1.0-alpha.19) (2022-01-22)

### Features

- **pintora-diagrams:** add mindmap support ([3062216](https://github.com/hikerpig/pintora/commit/306221660b34baad83d67435470d6aabc27b289f))

# [0.1.0-alpha.18](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.17...v0.1.0-alpha.18) (2022-01-16)

### Features

- **pintora-cli:** read cli args `--theme` ([a502c35](https://github.com/hikerpig/pintora/commit/a502c35cf1dbbad65c5847e2752e17711b8dccef))

# [0.1.0-alpha.11](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.10...v0.1.0-alpha.11) (2021-12-04)

### Features

- **pintora-diagrams:** Add basic activityDiagram [#3](https://github.com/hikerpig/pintora/issues/3) ([718a80e](https://github.com/hikerpig/pintora/commit/718a80e55d209d618fb855a463a151b880bf6fc3))

# [0.1.0-alpha.8](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.7...v0.1.0-alpha.8) (2021-08-29)

### Features

- [sequenceDiagram] add custom group color ([f93fe8a](https://github.com/hikerpig/pintora/commit/f93fe8a6fdb81af55eeb6f303f426c6150eeaf86))

# [0.1.0-alpha.4](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.3...v0.1.0-alpha.4) (2021-08-15)

### Bug Fixes

- **pintora-diagrams:** [componentDiagram] avoid interface node overlapping ([213cea0](https://github.com/hikerpig/pintora/commit/213cea090f40717390417e143366026d841f9808))

### Features

- **pintora-diagrams:** [erDiagram] add support for key coloumn like 'PK' ([64dad3d](https://github.com/hikerpig/pintora/commit/64dad3ddd8cefb912d7c83e14cf8a2dc0e64c5f8))

# [0.1.0-alpha.0](https://github.com/hikerpig/pintora/compare/v0.0.1...v0.1.0-alpha.0) (2021-08-01)

### Features

- Add componentDiagram ([c967f1c](https://github.com/hikerpig/pintora/commit/c967f1c9a969a813adac9a9967a06a024f04d73f))
