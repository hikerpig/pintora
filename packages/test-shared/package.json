{
  "name": "@pintora/test-shared",
  "version": "0.2.0-alpha.3",
  "description": "Shared testing utils for pintora",
  "keywords": [
    "pintora"
  ],
  "author": "hikerpig <hikerpigwinnie@gmail.com>",
  "homepage": "https://github.com/hikerpig/pintora#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "sideEffects": false,
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikerpig/pintora.git"
  },
  "scripts": {
    "compile": "tsc",
    "watch": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/hikerpig/pintora/issues"
  },
  "gitHead": "92814f05b3df654b588fe35e3af275a60ea61296"
}
