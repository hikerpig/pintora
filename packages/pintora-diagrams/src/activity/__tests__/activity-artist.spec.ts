import { diagramRegistry } from '@pintora/core'
import { EXAMPLES, stripStartEmptyLines } from '@pintora/test-shared'
import { testDraw, prepareDiagramConfig, stripDrawResultForSnapshot } from '../../__tests__/test-util'
import { activityDiagram } from '../index'

describe('activity-artist', () => {
  beforeAll(() => {
    prepareDiagramConfig()
    diagramRegistry.registerDiagram('activityDiagram', activityDiagram)
  })

  it('can draw', () => {
    expect(testDraw(EXAMPLES.activity.code).graphicIR.mark).toBeTruthy()
  })

  it('draw fork', () => {
    const code = stripStartEmptyLines(`
    activityDiagram
    start
    if (multiprocessor?) then
      fork
        :Action 1;
      forkagain
        :Action 2;
      endfork
      else (monoproc)
        :Action 1;
        :Action 2;
      endif
    end
    `)
    expect(stripDrawResultForSnapshot(testDraw(code))).toMatchSnapshot()
  })

  it('draw repeat', () => {
    const code = stripStartEmptyLines(`
    activityDiagram
    start
    repeat :prepare for each loop;
      :read data;
    repeatwhile (there is more data ?) is (alright then) not (nope)

    repeat
      :do something;
    repeatwhile (not done ?)
    end
    `)
    expect(stripDrawResultForSnapshot(testDraw(code))).toMatchSnapshot()
  })
})
