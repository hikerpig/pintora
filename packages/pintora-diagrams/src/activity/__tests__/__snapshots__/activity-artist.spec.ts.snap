// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity-artist draw fork 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {
          "id": "1",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "circle",
          },
        ],
        "class": "activity__keyword",
        "type": "group",
      },
      {
        "attrs": {
          "id": "2",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "class": "activity__decision-bg",
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "multiprocessor?",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "fill": "#fdb05e",
          "stroke": "#3b4044",
        },
        "class": "activity__condition-end",
        "type": "path",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "fill": "#3b4044",
              "radius": 2,
            },
            "type": "rect",
          },
          {
            "attrs": {},
            "children": [
              {
                "attrs": {
                  "id": "5",
                  "mark": [Circular],
                  "onLayout": [Function],
                },
                "children": [
                  {
                    "attrs": {
                      "fill": "#fdb05e",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  {
                    "attrs": {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 14,
                      "text": "Action 1",
                      "textAlign": "center",
                      "textBaseline": "middle",
                    },
                    "type": "text",
                  },
                ],
                "type": "group",
              },
            ],
            "type": "group",
          },
          {
            "attrs": {
              "id": "5",
              "mark": [Circular],
              "onLayout": [Function],
            },
            "children": [
              {
                "attrs": {
                  "fill": "#fdb05e",
                  "stroke": "#3b4044",
                },
                "type": "rect",
              },
              {
                "attrs": {
                  "fill": "#3b4044",
                  "fontFamily": "Source Code Pro, sans-serif",
                  "fontSize": 14,
                  "text": "Action 1",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "type": "text",
              },
            ],
            "type": "group",
          },
          {
            "attrs": {},
            "children": [
              {
                "attrs": {
                  "id": "7",
                  "mark": [Circular],
                  "onLayout": [Function],
                },
                "children": [
                  {
                    "attrs": {
                      "fill": "#fdb05e",
                      "stroke": "#3b4044",
                    },
                    "type": "rect",
                  },
                  {
                    "attrs": {
                      "fill": "#3b4044",
                      "fontFamily": "Source Code Pro, sans-serif",
                      "fontSize": 14,
                      "text": "Action 2",
                      "textAlign": "center",
                      "textBaseline": "middle",
                    },
                    "type": "text",
                  },
                ],
                "type": "group",
              },
            ],
            "type": "group",
          },
          {
            "attrs": {
              "id": "7",
              "mark": [Circular],
              "onLayout": [Function],
            },
            "children": [
              {
                "attrs": {
                  "fill": "#fdb05e",
                  "stroke": "#3b4044",
                },
                "type": "rect",
              },
              {
                "attrs": {
                  "fill": "#3b4044",
                  "fontFamily": "Source Code Pro, sans-serif",
                  "fontSize": 14,
                  "text": "Action 2",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "radius": 2,
        },
        "type": "rect",
      },
      {
        "attrs": {
          "id": "8",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "Action 1",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "id": "9",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "Action 2",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "id": "10",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "type": "circle",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "circle",
          },
        ],
        "class": "activity__keyword",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#fff",
              "opacity": 0.85,
            },
            "class": "label-bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "id": "2-3",
              "text": "yes",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "activity__edge-label",
            "type": "text",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#fff",
              "opacity": 0.85,
            },
            "class": "label-bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "id": "2-8",
              "text": "no",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "activity__edge-label",
            "type": "text",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;

exports[`activity-artist draw repeat 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {
          "id": "1",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "circle",
          },
        ],
        "class": "activity__keyword",
        "type": "group",
      },
      {
        "attrs": {
          "id": "2-end",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "class": "activity__decision-bg",
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "there is more data ?",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "id": "2",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "prepare for each loop",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "activity__repeat-start",
        "type": "group",
      },
      {
        "attrs": {
          "id": "3",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "read data",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "id": "4-end",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "class": "activity__decision-bg",
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "not done ?",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "fill": "#fdb05e",
          "stroke": "#3b4044",
        },
        "class": "activity__repeat-start",
        "type": "path",
      },
      {
        "attrs": {
          "id": "5",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "text": "do something",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "type": "group",
      },
      {
        "attrs": {
          "id": "6",
          "mark": [Circular],
          "onLayout": [Function],
        },
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "type": "circle",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "circle",
          },
        ],
        "class": "activity__keyword",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#fff",
              "opacity": 0.85,
            },
            "class": "label-bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "id": "2-end-2",
              "text": "alright then",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "activity__edge-label",
            "type": "text",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#fff",
              "opacity": 0.85,
            },
            "class": "label-bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "id": "2-end-4",
              "text": "nope",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "activity__edge-label",
            "type": "text",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "lineJoin": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#fff",
              "opacity": 0.85,
            },
            "class": "label-bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 14,
              "id": "4-end-6",
              "text": "no",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "class": "activity__edge-label",
            "type": "text",
          },
          {
            "attrs": {
              "lineCap": "round",
              "stroke": "#fdb05e",
            },
            "type": "path",
          },
        ],
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;
